src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.cs

```diff
@@ -184,6 +184,13 @@ internal static BoundBlock ConstructAutoPropertyAccessorBody(SourceMemberMethodS
             }
 
             var field = property.BackingField;
+            if (field is null)
+            {
+                // This happens for public int { set; } where we produce ERR_AutoPropertyMustHaveGetAccessor
+                Debug.Assert(!property.HasGetAccessor);
```

Comment:
>property.HasGetAccessor [](http://example.com/codeflow?start=30&length=23)

There is an existing API to check for a setter. We don't need to expose a new one.